bandit/plugins/general_bad_file_permissions.py

```diff
@@ -73,27 +83,35 @@ def _stat_is_dangerous(mode):
 @test.test_id("B103")
 def set_bad_file_permissions(context):
     if "chmod" in context.call_function_name:
-        if context.call_args_count == 2:
+        if (
+                context.call_function_name_qual.startswith("os.")
+                and context.call_args_count == 2
+        ):  # os chmod
+            filename = context.get_call_arg_at_position(0)
             mode = context.get_call_arg_at_position(1)
+        elif context.call_args_count == 1:  # pathlib chmod
```

Comment:
So this may result in some false positives. Any function named chmod with one arg will end up here. While, I can think of some other cases in other checks as well, there's not a better way to handle this at the moment. Ideally a symbol table that tracks instances of certain types would result in more accuracy.

Reply:
Might be quite a few false positives on any function containing "chmod". Not sure how prevalent this would be, but function names like "lunchmode", "chmodulus", "chmodel", etc

Reply:
Unfortunately even the current version of this test is not restrictive enough. The following code is enough to trigger a high severity issue in the current version of bandit:

```python
def chmod(a, b):
    pass

chmod("file.txt", 0x777)
```

I've fixed this by checking the fully qualified name. But as you say, its trickier to solve in the case of pathlib.

I see a few possibilities:
1. Accept that some false positives could occur - this would already be the case for the current implementation anyway (although now it would be cases of 1 arg functions called chmod as opposed to 2 arg functions)
2. Reduce the confidence (either to medium or low) for the new case.
3. Put this PR on hold and do it properly, noting that the current bandit core is somewhat limiting in this regard.

I see (3) as a longer-term goal because it would be needed as more advanced tests are added.

In the meantime, I've also added a missing check that `pathlib` is actually in the current context, which should help reduce false positives as well.

Reply:
> Might be quite a few false positives on any function containing "chmod". Not sure how prevalent this would be, but function names like "lunchmode", "chmodulus", "chmodel", etc

In addition, we could update the check so that the `call_function_name` is explicitly one of `chmod`, `lchmod`, `fchmod`. That will at least narrow down the scope for false positives.